jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -2585,19 +2584,18 @@ public ClassLoader getClassLoader()
 
             // no security manager just return the classloader
             if (!_usingSecurityManager)
+            {
                 return _classLoader;
+            }
             else
             {
                 // check to see if the classloader of the caller is the same as the context
-                // classloader, or a parent of it
+                // classloader, or a parent of it, as required by the javadoc specification.
                 try
                 {
-                    Class<?> reflect = Loader.loadClass("sun.reflect.Reflection");
-                    Method getCallerClass = reflect.getMethod("getCallerClass",Integer.TYPE);
-                    Class<?> caller = (Class<?>)getCallerClass.invoke(null,2);
-
+                    Class<?> caller = new Caller().getCallerClass(2);
```

Comment:
"new Caller()" should be wrapped in an `java.security.AccessController.doPrivileged(..)` block..

Reply:
Will do, but we're going into a rabbit hole here - Jetty has not been coded with a SecurityManager in mind.